### PR TITLE
Update MaskedLMHead to support dtype=bfloat16/float16/float64

### PR DESCRIPTION
**Pull request by g-dspencer**
- Update MaskedLMHead to support dtype=bfloat16/float16/float64.

Inspired by https://github.com/keras-team/keras/commit/397ad5771574dd96e8aa47c5d52735ad860ecd0f
i.e. using the idiom (?) of `dtype=self._dtype_policy`.

This is to fix https://github.com/keras-team/keras-nlp/issues/1195

I had a previous try at this where I accidentally included print statements, sorry.

### CODE REVIEW

File: keras_nlp/layers/modeling/masked_lm_head_test.py [hunk 1]
```diff
@@ -119,6 +146,32 @@ def test_one_train_step(self):
         loss = model.train_on_batch(x=(token_data, position_data), y=label_data)
         self.assertGreater(loss, 0)
 
+    @parameterized.named_parameters(
```
Comment (mattdangerw) [inline keras_nlp/layers/modeling/masked_lm_head_test.py:149]: I would kill this test. Compiling a real loss function can make for slower tests, and with the parameterized testing this could slow down our suite.

**Review by mattdangerw — COMMENTED** [review]
Thanks!

File: keras_nlp/layers/modeling/masked_lm_head_test.py [hunk 2]
```diff
@@ -36,6 +39,30 @@ def test_valid_call(self):
...
+    )
+    def test_valid_call_with_dtype(self, dtype):
+        head = masked_lm_head.MaskedLMHead(
+            vocabulary_size=100,
+            activation="softmax",
+            dtype=dtype,
+        )
+        encoded_tokens = keras.Input(shape=(10, 16))
+        positions = keras.Input(shape=(5,), dtype="int32")
+        outputs = head(encoded_tokens, masked_positions=positions)
```
Comment (mattdangerw) [inline keras_nlp/layers/modeling/masked_lm_head_test.py:56]: This might need a rebase over master. This should be `mask_positions` now. This is causing a lot of test failures.

**Comment by sachinprasadhs** [comment]
@g-dspencer , Is this PR still relevant, if yes, could you please update the status of this PR. Thanks
